stdlib/http/src/main/ballerina/http/service_endpoint.bal

```diff
@@ -165,12 +231,184 @@ public type KeepAlive "AUTO"|"ALWAYS"|"NEVER";
 
 function Listener::init (ServiceEndpointConfiguration c) {
     self.config = c;
+    match self.config.authProviders {
+        AuthProvider[] providers => addAuthFiltersForSecureListener(self.config, self.instanceId);
+        () => {}
+    }
     var err = self.initEndpoint();
     if (err != null) {
         throw err;
     }
 }
 
+# Add authn and authz filters
+#
+# + config - `ServiceEndpointConfiguration` instance
+# + instanceId - Endpoint instance id
+function addAuthFiltersForSecureListener(ServiceEndpointConfiguration config, string instanceId) {
+    // add authentication and authorization filters as the first two filters.
+    // if there are any other filters specified, those should be added after the authn and authz filters.
+    if (config.filters == null) {
+        // can add authn and authz filters directly
+        config.filters = createAuthFiltersForSecureListener(config, instanceId);
+    } else {
+        Filter[] newFilters = createAuthFiltersForSecureListener(config, instanceId);
+        // add existing filters next
+        int i = 0;
+        while (i < lengthof config.filters) {
+            newFilters[i + (lengthof newFilters)] = config.filters[i];
+            i = i + 1;
+        }
+        config.filters = newFilters;
+    }
+}
+
+# Create an array of auth and authz filters.
+#
+# + config - `ServiceEndpointConfiguration` instance
+# + instanceId - Endpoint instance id
+# + return - Array of Filters comprising of authn and authz Filters
+function createAuthFiltersForSecureListener(ServiceEndpointConfiguration config, string instanceId) returns (Filter[]) {
```

Comment:
Since we know for sure that this logic is now executed if `AuthProviders` exist. Can we remove `ServiceEndpointConfiguration` here and pass `AuthProvider[]`. That will simplicy the logic and get rig of `match` blocks.

Reply:
The 

```
() => {
            // No auth providers configured.
        }
```

condition will never get executed here.

Reply:
Yes, agree on that. But the `ServiceEndpointConfiguration` may be used for future as well. So that the `config` parameter is passed and simplify the function implementation with https://github.com/ballerina-platform/ballerina-lang/pull/11173/commits/af56a725cbaeaef64ae4342347c2e21d96b63af7